src/test/java/com/google/sps/DetectTextFromImageTest.java

```diff
@@ -0,0 +1,150 @@
+
+  private final String IMAGE_BASE_URI =
+      "https://shop-by-photos-step-2020.ey.r.appspot.com/get-image-url?blob-key=";
+  private final Feature TEXT_DETECTION_FEATURE = Feature.newBuilder().setType(Feature.Type.TEXT_DETECTION).build();
+
+  private String nullImageKey = "";
+  private String invalidImageKey = "abc £$%  d\n\n\n";
+  // This key is copied from Cloud Datastore. Make sure it exists there before running the tests.
+  private String validImageKey =
+      "AMIfv943BzZrlhyLIN_el9l15Zz2LUp4H3bRNwHj2yvPaDNuv0uZjsp8vSySw0u5pEp-sspSVv4U5fkCT6SY1vbTeObItcslXvPswVieK3rInnCc0nBkTDlfIgWqxlvYjFVPS1QQqCjiY7NlRNSHA3gRXtsv6hrj7_J3_c_DaCP4jhpbZid2bkKvOA1XD0geHOXbCjKHf0ZIbEU4wKqQgGcCxT0X4Ddi4o6Nk8rhN7CNbN27GD-iBfrIG2RwNXr24xLLlFg8Bw9xCUxrbaXO6tkJNHndSghb0x4KAZ-IEPYKHfuWtLfXSmgMV_D1hHZdqShTaUZIyJu-";
+
+  private List<String> nullShoppingList = new ArrayList<>();
+  private List<String> validShoppingList = new ArrayList<>();
+  private List<String> invalidShoppingList = new ArrayList<>();
+
+  private DetectTextFromImage detectText;
+
+  @Before
+  public void setUp() {
+    validShoppingList.add("Text:Bag ");
+    validShoppingList.add(
+        "Position : vertices { x: 151 y: 94 } vertices { x: 187 y: 94 } vertices { x: 187 y: 115 } vertices { x: 151 y: 115 }");
+    validShoppingList.add("Text:Bag");
+    validShoppingList.add(
+        "Position : vertices { x: 152 y: 94 } vertices { x: 187 y: 96 } vertices { x: 186 y: 114 } vertices { x: 151 y: 112 }");
+
+    invalidShoppingList.add("!%^&*+");
+  }
+
+  /**
+   * ********************************************** 
+   * **** TESTS FOR DetectTextFromImage constructor
+   * **********************************************
+   */
+  @Test
+  public void DetectTextFromImage_forNullImageKey() {
+    Assertions.assertThrows(
+        PhotoShoppingException.class,
+        () -> {
+            new DetectTextFromImage(nullImageKey);
+        });
+  }
+
+  @Test
+  public void DetectTextFromImage_forInvalidImageKey() {
+    Assertions.assertThrows(
+        PhotoShoppingException.class,
+        () -> {
+            new DetectTextFromImage(invalidImageKey);
+        });
+  }
+
+  /**
+   * ******************************************** 
+   * **** TESTS FOR shoppingImageRequestGenerator
+   * ********************************************
+   */
+  // Testing the shoppingImageRequestGenerator routine with valid Image source
+  @Test
+  public void shoppingImageRequestGenerator_forValidImageKey() {
+    String expectedUri = IMAGE_BASE_URI + validImageKey;
+    try {
+        DetectTextFromImage detectText = new DetectTextFromImage(validImageKey);
+        List<AnnotateImageRequest> actual_requests = detectText.shoppingImageRequestGenerator();
+        Assert.assertEquals(expectedUri, actual_requests.get(0).getImage().getSource().getImageUri());
+    } catch (PhotoShoppingException e) {
+        Assert.fail("Not expected: \nThrew PhotoShoppingException for Invalid Key");
+    }
```

Comment:
Normally there should be no "try-catch" in a test because if such an unexcepted exception happens, the test will fail directly.